pkg/asset/agent/image/ignition.go

```diff
@@ -108,10 +116,28 @@ func (a *Ignition) Generate(dependencies asset.Parents) error {
 		return err
 	}
 
+	agentMirror := &mirror.AgentMirror{}
+	dependencies.Get(agentMirror)
+
+	mirrorRegistriesMount := ""
+	if _, err := os.Stat(mirror.RegistriesConfFilename); err == nil {
+		mirrorRegistriesMount = fmt.Sprintf("-v %s:/etc/containers/registries.conf:z", filepath.Join("/etc/assisted", mirror.RegistriesConfFilename))
+	}
+	releaseImageMirror := ""
+	for _, config := range agentMirror.MirrorConfig {
+		// Use the source location not including the version
+		source := strings.Split(":", agentManifests.ClusterImageSet.Spec.ReleaseImage)
+		if config.Location == source[0] {
```

Comment:
Could this also be a validation? To check that if MirrorConfigs are provided, at least one of the MirrorConfigs matches the ClusterImageSet's ReleaseImage?

Reply:
Good point, I can get the ClusterImageSet info in the RegistriesConf asset and add that check.

Reply:
Actually I can't add the validation in the RegistriesConf asset since I can't get access to another asset (i.e. ClusterImageSet) in the Load() but I should be able to add the validation in the mirror asset as I can add ClusterImageSet as a dependency there. I will follow up in a separate PR with the validation.